sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClient.java

```diff
@@ -910,18 +911,26 @@ Mono<VerifyResult> verifyData(SignatureAlgorithm algorithm, byte[] data, byte[]
         });
     }
 
-    private Mono<Boolean> ensureValidKeyAvailable() {
+    private Mono<Boolean> isValidKeyLocallyAvailable() {
         boolean keyNotAvailable = (key == null && keyCollection != null);
-        boolean keyNotValid = (key != null && !key.isValid());
 
-        if (keyNotAvailable || keyNotValid) {
+        if (keyNotAvailable) {
             if (keyCollection.equals(CryptographyClientImpl.SECRETS_COLLECTION)) {
                 return getSecretKey().map(jsonWebKey -> {
-                    key = (jsonWebKey);
+                    key = jsonWebKey;
 
                     if (key.isValid()) {
                         if (localKeyCryptographyClient == null) {
-                            localKeyCryptographyClient = initializeCryptoClient(key, implClient);
+                            try {
+                                localKeyCryptographyClient = initializeCryptoClient(key, implClient);
+                            } catch (RuntimeException e) {
+                                localOperationNotSupported = true;
+
+                                LOGGER.logExceptionAsWarning(
+                                    new RuntimeException("Defaulting to service use for cryptographic operations.", e));
```

Comment:
Does the exception that gets thrown here (`e`) include useful information on why we are defaulting to using the service?

Reply:
The default behavior is to fetch a key from Key Vault and perform crypto operations locally and, since the client was not able to perform an operation locally and is going back to the service, I thought it appropriate to let users know. Should we move this message to INFO or DEBUG?